### ISSUE TITLE
http-client incomaptible with php 7.x

### ISSUE THREAD

**Issue by tomasz-kusy** [symfony/symfony#42221]
Hi,

http-client, branch (at least) 5.3

https://github.com/symfony/http-client/commit/29ff82db6fbc1ae51d4e7e88b161c68573f22555
"Leverage str_contains/str_starts_with Leverage str_ends_with"
These functions are available starting from PHP8.
But in composer.json is still:
`"require": {
        "php": ">=7.2.5",`

**Comment by stof** [comment]
There is no issue. the same commit adds `symfony/polyfill-php80` in the dependencies.

**Comment by tomasz-kusy** [comment]
Thanks for your reply. As I understand when I'm use this package within Symfony framework everything is ok. But without framework (as I use) the polyfill bootstrap.php never is called and it breaks BC without warning.

**Comment by stof** [comment]
Well, Symfony is not the one calling the `bootstrap.php` file. It is part of the config for the composer autoloading.

If your own stack does not load `vendor/autoload.php` to use the composer-generated autoloader but uses its own autoloading stack instead, you might indeed face issues if your own autoloading logic is not equivalent to the composer one. But that's an issue on your side then. The only official way to install Symfony packages is through composer.

**Comment by tomasz-kusy** [comment]
I use composer autoloader.
```
Package operations: 0 installs, 1 update, 1 removal
  - Removing symfony/polyfill-php80 (v1.23.0)
  - Upgrading symfony/http-client (v5.3.3 => 5.4.x-dev 2c33b72): Extracting archive
  ```
  Branch `5.4`
  Why `http-client` "provides" `polyfill-php80`?
  https://github.com/symfony/http-client/blob/29ff82db6fbc1ae51d4e7e88b161c68573f22555/composer.json#L22
  ```
  "provide": {
        "php-http/async-client-implementation": "*",
        "php-http/client-implementation": "*",
        "psr/http-client-implementation": "1.0",
        "symfony/http-client-implementation": "2.4",
        "symfony/polyfill-php80": "^1.16"
    },
   ```

**Comment by nicolas-grekas** [comment]
Looks like a typo, fixed in 855232fff0
Give me a few minutes to propagate this to packagist and let me know if it's better.
